python/lsst/cp/pipe/ptc.py

```diff
@@ -718,24 +809,18 @@ def errFunc(p, x, y):
 
             # Non-linearity residuals (NL of mean vs time curve): percentage, and fit to a quadratic function
             # In this case, len(parsIniNl) = 3 indicates that we want a quadratic fit
-            parsIniNl = [1., 1., 1.]
-            if self.config.doFitBootstrap:
-                parsFit, parsFitErr = self._fitBootstrap(parsIniNl, timeVecFinal, meanVecFinal,
-                                                         self.funcPolynomial)
-            else:
-                parsFit, parsFitErr = self._fitLeastSq(parsIniNl, timeVecFinal, meanVecFinal,
-                                                       self.funcPolynomial)
-            linResidualTimeIndex = self.config.linResidualTimeIndex
-            if timeVecFinal[linResidualTimeIndex] == 0.0:
-                raise RuntimeError("Reference time for linearity residual can't be 0.0")
-            linResidual = 100*(1 - ((meanVecFinal[linResidualTimeIndex] /
-                                     timeVecFinal[linResidualTimeIndex]) / (meanVecFinal/timeVecFinal)))
-
-            dataset.nonLinearity[ampName] = parsFit
-            dataset.nonLinearityError[ampName] = parsFitErr
-            dataset.nonLinearityResiduals[ampName] = linResidual
-
-        return dataset
+
+            (c0, linearizerTableRow, linResidualNl, parsFitNl,
+             parsFitErrNl) = self.calculateLinearityResidualAndLinearizers(timeVecFinal, meanVecFinal)
+            # LinearizerLookupTable
+            tableArray[i, :] = linearizerTableRow
+
+            dataset.nonLinearity[ampName] = parsFitNl
+            dataset.nonLinearityError[ampName] = parsFitErrNl
+            dataset.nonLinearityResiduals[ampName] = linResidualNl
+            dataset.coefficientLinearizeSquared[ampName] = c0
```

Comment:
I still don't know how the `c0` will be read in case that someone wants to use `LinearizeSquared`, and hence I don't know where it should be exactly stored and in which format We discussed this about in one of our meetings, I think, and I believe we agreed that, for the moment, we would just save it somewhere. Since the `dataset` structure was already in place, I decided to append the coefficient to it.   I have used the PTC data, such as the gains, and the noise, to do comparisons with other codes (e.g., the BOT analyses), and pickles have worked fine so far.

Reply:
Since writing this, I've discovered that yaml files aren't fully supported by butler yet (particularly in gen3).  I am going to try to get that working soon, because I like having human readable things easily readable by humans.  Pickle is fine for now.

Reply:
Can you let me know what you mean by "not working". I use yaml in gen3 all the time.

Reply:
I've had to add storageClass/formatter definitions to get yaml dict output to be understood (and I thought I'd rebased to ensure I had everything up to date).

Reply:
Ah, well, we may want to move this definition out of the daf_butler tests then:
```yaml
storageClasses:
  StructuredDataDictYaml:
    pytype: dict
```
and add
```yaml
StructuredDataDictYaml: lsst.daf.butler.formatters.yamlFormatter.YamlFormatter
```
to the `config/datastores/formatter.yaml`.  We hadn't enabled simply python types by default and expected there to be more expressive types read from YAML files. The default YamlFormatter isn't super great at casting the dict returned to a new type (it does try and could be improved).